packages/mongodb-rag-ingest/src/commands/all.test.ts

```diff
@@ -142,4 +131,78 @@ describe("allCommand", () => {
       await ingestMetaStore.close();
     }
   });
+
+  it("cleans up pages and embedded content that are no longer in the data sources", async () => {
+    const ingestMetaStore = makeIngestMetaStore({
+      connectionUri: uri,
+      databaseName,
+      entryId: "all",
+    });
+
+    // Mock data sources
+    const mockDataSources: DataSource[] = [
+      { name: "source1", fetchPages: () => Promise.resolve([]) },
+      { name: "source2", fetchPages: () => Promise.resolve([]) },
+    ];
+
+    // Insert mock pages and embedded content
+    assert(mongoClient)
+    await mongoClient
+      .db(databaseName)
+      .collection("pages")
+      .insertMany([
+        { id: "page1", sourceName: "source1", content: "content1" },
+        { id: "page2", sourceName: "source2", content: "content2" },
+        { id: "page3", sourceName: "source3", content: "content3" },
+      ]);
+    await mongoClient
+      ?.db(databaseName)
+      .collection("embedded_content")
+      .insertMany([
+        { id: "embed1", sourceName: "source1", embedding: [1, 2, 3] },
+        { id: "embed2", sourceName: "source2", embedding: [4, 5, 6] },
+        { id: "embed3", sourceName: "source3", embedding: [7, 8, 9] },
+      ]);
```

Comment:
rather than using the mongoclient here? could you just use methods that use data sources? 

so like you could do use the `doAll` command to insert 2 data sources as a set up, and then you could use doAll command again with 1 data source and validate that the other was deleted.